numpy/core/src/multiarray/calculation.c

```diff
@@ -1111,15 +1123,17 @@ PyArray_Clip(PyArrayObject *self, PyObject *min, PyObject *max, PyArrayObject *o
 
     /*
      * Do we still not have a suitable output array?
-     * Create one, now
+     * Create one, now. Now matter why the array is not suitable a copy has
+     * to be made. This may be just to avoid memory overlap though.
      */
     if (!outgood) {
         int oflags;
-        if (PyArray_ISFORTRAN(out))
+        if (PyArray_ISFORTRAN(self))
             oflags = NPY_ARRAY_FARRAY;
         else
             oflags = NPY_ARRAY_CARRAY;
-        oflags |= NPY_ARRAY_WRITEBACKIFCOPY | NPY_ARRAY_FORCECAST;
+        oflags |= (NPY_ARRAY_WRITEBACKIFCOPY | NPY_ARRAY_FORCECAST |
+                   NPY_ARRAY_ENSURECOPY);
         Py_INCREF(indescr);
         newout = (PyArrayObject*)PyArray_FromArray(out, indescr, oflags);
```

Comment:
Actually, will think a minute about this. Forcing the copy is somewhat correct, but actually... The new array might as well be uninitialized here...
But might end up being lazy and just fix the small stuff :).

Reply:
OK, was lazy here... Ideally we will make this a real ufunc anyway. Would have to read up what the problem with that was, maybe we should also just make it a ufunc and dispatch either to minimum, maximum or the internal ufunc (because of the omission arguments) and ignore the `fastclip` function on the dtype....

Reply:
I think the main problem in making functions like this ufuncs are that the arguments don't fit neatly into the usual ufunc signature.